pkg/controllers/v1alpha1/databackup/implement.go

```diff
@@ -18,490 +18,162 @@ package databackup
 
 import (
 	"context"
-	"errors"
 	"fmt"
-	"os"
-	"reflect"
-	"strconv"
-	"strings"
-	"time"
-
-	"github.com/go-logr/logr"
-	"gopkg.in/yaml.v2"
+	"github.com/fluid-cloudnative/fluid/pkg/dataoperation"
+	"github.com/fluid-cloudnative/fluid/pkg/runtime"
+	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/client-go/tools/record"
-	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
 
 	"github.com/fluid-cloudnative/fluid/api/v1alpha1"
 	"github.com/fluid-cloudnative/fluid/pkg/common"
 	cdatabackup "github.com/fluid-cloudnative/fluid/pkg/databackup"
-	"github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/operations"
-	goosefs "github.com/fluid-cloudnative/fluid/pkg/ddc/goosefs/operations"
 	"github.com/fluid-cloudnative/fluid/pkg/utils"
-	"github.com/fluid-cloudnative/fluid/pkg/utils/docker"
-	"github.com/fluid-cloudnative/fluid/pkg/utils/helm"
-	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
 )
 
-// DataBackupReconcilerImplement implements the actual reconciliation logic of DataBackupReconciler
-type DataBackupReconcilerImplement struct {
-	client.Client
-	Log      logr.Logger
-	Recorder record.EventRecorder
-	*DataBackupReconciler
-}
-
-// NewDataBackupReconcilerImplement returns a DataBackupReconcilerImplement
-func NewDataBackupReconcilerImplement(client client.Client, log logr.Logger, recorder record.EventRecorder, databackupReconciler *DataBackupReconciler) *DataBackupReconcilerImplement {
-	r := &DataBackupReconcilerImplement{
-		Client:               client,
-		Log:                  log,
-		Recorder:             recorder,
-		DataBackupReconciler: databackupReconciler,
-	}
-	return r
-}
-
-// ReconcileDataBackupDeletion reconciles the deletion of the DataBackup
-func (r *DataBackupReconcilerImplement) ReconcileDataBackupDeletion(ctx reconcileRequestContext) (ctrl.Result, error) {
-	log := ctx.Log.WithName("ReconcileDataBackupDeletion")
-
-	// 1. Delete release if exists
-	releaseName := utils.GetDataBackupReleaseName(ctx.DataBackup.Name)
-	err := helm.DeleteReleaseIfExists(releaseName, ctx.DataBackup.Namespace)
-	if err != nil {
-		log.Error(err, "can't delete release", "releaseName", releaseName)
-		return utils.RequeueIfError(err)
-	}
-
-	// 2. Release lock on target dataset if necessary
-	err = r.releaseLockOnTargetDataset(ctx, log)
+// UpdateStatusByHelmStatus update the operation status according to helm job status
+func (r *DataBackupReconciler) UpdateStatusByHelmStatus(ctx runtime.ReconcileRequestContext, object client.Object, opStatus *v1alpha1.OperationStatus) (err error) {
```

Comment:
I'm little confused about why the function is called `ByHelmStatus`? It seems it gets status from pod.Status.

Reply:
`ByHelmPodStatus` or just `ByPodStatus` ？